sdks/go/pkg/beam/io/fhirio/read_test.go

```diff
@@ -0,0 +1,84 @@
+// Licensed to the Apache Software Foundation (ASF) under one or more
+// contributor license agreements.  See the NOTICE file distributed with
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package fhirio
+
+import (
+	"errors"
+	"net/http"
+	"strings"
+	"testing"
+
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/testing/passert"
+	"github.com/apache/beam/sdks/v2/go/pkg/beam/testing/ptest"
+)
+
+func TestRead(t *testing.T) {
+	testCases := []struct {
+		name           string
+		client         fhirStoreClient
+		containedError string
+	}{
+		{
+			name: "Read Request Failed",
+			client: &fakeFhirStoreClient{
+				fakeReadResources: func(resource string) (*http.Response, error) {
+					return nil, errors.New("")
+				},
+			},
+			containedError: "Failed to fetch resource",
```

Comment:
I don't see a problem with comparing error strings, in fact frequently find tests that only check errors are returned but not contents are testing a different error path than the test name says. Creating custom error types is probably more elaborate than needed right now. Since we're only checking the error contains a substring (instead of checking the whole error), I think this isn't too bad, we should just keep comparing the strings as-is.

Reply:
I had considered having the error messages as unexported constants. But in this particular test, the error messages contain formatting. So we'd have to properly format it, which would be a bit tricky because some error messages have more formatting arguments than others + the individual formatting arguments aren't easily available at assertion time.
I think custom error types would be interesting, but maybe at a later stage?